plugins/vpc-shared-eni/network/bridge_windows.go

```diff
@@ -22,6 +22,7 @@ import (
 	"github.com/aws/amazon-vpc-cni-plugins/network/vpc"
 
 	"github.com/Microsoft/hcsshim"
+	"github.com/Microsoft/hcsshim/hcn"
```

Comment:
Most of the new code in this package already exists in ecs-eni plugin. Have we considered creating a shared HNS package under network/hns that both plugins call import, instead of duplicating it?

I am not suggesting we do so (yet), just trying to understand what the implications would be, and whether it is worth doing it now. I know we had discussed this before when developing ecs-eni plugin, but opted not to do so at the time because we didn't know how many callers of such a package would be, and also because of time constraints.

Reply:
I agree with you that such a refactoring would be beneficial for us in the long term. 

Unlike last time where we tried to abstract complete HNS package, we can refactor only the specific methods which distinguish between HNS V1 and HNS V2. The caller would still have controls to implement any additional logic required while invoking HNS V1 vs V2 APIs.

I was thinking something on the lines of https://github.com/rawahars/amazon-vpc-cni-plugins/pull/3
This is a sample PR to showcase what the refactoring would look like for both the plugins.

Thoughts on this one?